### PR TITLE
Test for daos

### PR DESCRIPTION
**Pull request by AnonymousWalker**
This change is [Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/653)

### CODE REVIEW

**Review by jsarabia — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/653#-:-N9mOh5q0q6gsJ5RdJ_k:bnfp4nl)* status: 0 of 3 files reviewed, all discussions resolved

**Comment by AnonymousWalker** [comment]
Moved to #670 due to git issues

File: jvm/workbookapp/src/integration-test/kotlin/integrationtest/persistence/repository/TestCollectionRepository.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    @Inject lateinit var collectionRepository: ICollectionRepository
+
+    init {
+        DaggerTestPersistenceComponent.create().inject(this)
+    }
+
+    private val db = dbEnvProvider.get()
+
+    @Test
+    fun testDeleteResource() {
```
Comment (PurpleGuitar) [inline jvm/workbookapp/src/integration-test/kotlin/integrationtest/persistence/repository/TestCollectionRepository.kt:46]: When writing a complex test like this one, it's good practice to include some prose at the top that describes what the test is doing.  What is the object under test?  What's the expected behavior?

In this case, why are there six takes before the delete statement, but three after?  How do I know whether this is correct behavior, and why?

File: jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/dao/AudioPluginDaoTest.kt [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+        Assert.assertEquals(1, dao.fetchAll().size)
+        Assert.assertEquals(id1, id2)
+    }
+
+    @Test
+    fun testInsertThrowsException() {
+        try {
+            dao.insert(defaultEnity.copy(id = 1))
+            Assert.fail("Inserting nonzero id should throw an exception")
```
Comment (PurpleGuitar) [inline jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/dao/AudioPluginDaoTest.kt:71]: Good use of failure message.  It's a good idea to use messages on other asserts too, e.g. `assertEquals(1, dao.fetchAll().size, "After inserting, there should be one record.")`

File: jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/dao/AudioPluginDaoTest.kt [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        dao.update(updated)
+
+        Assert.assertEquals(
+            updated,
+            dao.fetchById(insertedId)
+        )
+    }
+
+    @Test
+    fun `test update duplicated values throws exception`() {
```
Comment (PurpleGuitar) [inline jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/dao/AudioPluginDaoTest.kt:111]: I like how descriptive this function name is, but it's inconsistent with the other test functions in this class.  For readability and maintainability, the test function names should follow the same pattern.

File: jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/dao/PreferenceDaoTest.kt [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    fun testUpsert() {
+        val pref = PreferenceEntity("k3", "v3")
+        Assert.assertNull(dao.fetchByKey(pref.key))
+
+        // insert
+        dao.upsert(pref)
+
+        var result = dao.fetchByKey(pref.key)
+        Assert.assertNotNull(result)
+        Assert.assertEquals(pref, result)
```
Comment (PurpleGuitar) [inline jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/dao/PreferenceDaoTest.kt:91]: It's probably worth also testing that there is one more object in the database than there was before.

File: jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/dao/PreferenceDaoTest.kt [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+        // update
+        val newValue = "v3-updated"
+        dao.upsert(pref.copy(value = newValue))
+
+        result = dao.fetchByKey(pref.key)
+        Assert.assertNotNull(result)
+        Assert.assertEquals(newValue, result!!.value)
```
Comment (PurpleGuitar) [inline jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/dao/PreferenceDaoTest.kt:99]: It's probably worth also testing that the number of objects in the database has not changed.

File: jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/dao/PreferenceDaoTest.kt [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    }
+
+    @Test
+    fun testDelete() {
+        val entity = dao.fetchByKey(prefs[0].key)
+        Assert.assertNotNull(entity)
+
+        dao.delete(entity!!)
+
+        Assert.assertNull(dao.fetchByKey(entity.key))
```
Comment (PurpleGuitar) [inline jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/dao/PreferenceDaoTest.kt:109]: It's probably worth testing that the number of objects in the database has decreased.

File: jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/dao/PreferenceDaoTest.kt (hunk 4 above)
Comment (AnonymousWalker) [reply jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/persistence/dao/PreferenceDaoTest.kt:91]: thanks for the catch. Although the APIs of this particular dao (PreferenceDao) do not support `fetchAll()`, I am happy to manually check it with the database query.

**Comment by AnonymousWalker** [comment]
@PurpleGuitar Thank you, I have made the changes in #670, since this one was closed due to git issues.

**Pull request closed without merging**